### ISSUE TITLE
Can use oauth-handler  on other host?

### ISSUE THREAD

**Issue by alitrack** [duplicati/duplicati#2521]
I have:
- [ ] searched open and closed issues for duplicates

----------------------------------------

### Version info

**Duplicati Version:** 2.0.1.64
**Operating System:**  Windows 2008 R2 & Mac OS X
**Backend:**

### Bug description
Describe here the issue that you are experiencing.
because appspot.com (GAE) is blocked by CN, so I have to use it over proxy.

### Steps to reproduce
- using hyphens as bullet points
- list the steps
- that reproduce the bug

**Actual result:** Describe here what happens after you run the steps above (i.e. the buggy behaviour)
**Expected result:** Describe here what should happen after you run the steps above (i.e. what would be the correct behaviour)

**Comment by kenkendk** [comment]
There is not currently another service than the GAE version.

How does the block work, is it DNS level? If so, I can put up another DNS record.
If the block is IP, I would need to host a proxy server that passes requests through.

**Comment by alitrack** [comment]
seems DNS level, but most IP are blocked too.

any way, appspot or other Google service is not easy to access when in China.

and how about modify and deploy it on heroku.com?

btw, do you know [rclone](https://github.com/ncw/rclone), another similar project.

**Comment by kenkendk** [comment]
Yes, I know about rclone and acd_cli, they both seem to go towards using GAE as well.

**Comment by alitrack** [comment]
rclone uses local server to do oauth.
[oauthutil.go](https://github.com/ncw/rclone/blob/52e1bfae2a8b67b0056af81eac00a0c394915121/oauthutil/oauthutil.go)

**Comment by kenkendk** [comment]
Yes, I know you can run a local oauth-service, but that requires you have the ability to open a connection towards the internet. I think such a solution is better solved in a separate project "local oauth server". If you want this solution, you can use the Duplicati OAuth code and run it locally with the GAE SDK:
https://github.com/duplicati/oauth-handler

**Comment by kenkendk** [comment]
This is actually duplicate of #2470

**Comment by duplicatibot** [comment]
This issue has been mentioned on **Duplicati**. There might be relevant details there:

https://forum.duplicati.com/t/running-the-oauth-handler-in-aws/11122/2
